From abf0026782c0ba4643feb25ded022e8c12725584 Mon Sep 17 00:00:00 2001
From: Victor Seva <linuxmaniac@torreviejawireless.org>
Date: Mon, 9 Sep 2013 16:52:47 +0200
Subject: [PATCH] uac_redirect: get_redirect() check reason value before using
 it.

This fix a core dump when get_redirects() config fuction is called with just
the one parameter.
---
 modules/uac_redirect/rd_funcs.c |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/modules/uac_redirect/rd_funcs.c b/modules/uac_redirect/rd_funcs.c
index ca34e0c..aa8dbd0 100644
--- a/modules/uac_redirect/rd_funcs.c
+++ b/modules/uac_redirect/rd_funcs.c
@@ -96,9 +96,12 @@ int get_redirect( struct sip_msg *msg , int maxt, int maxb,
 		}
 		if (max==0)
 			continue;
-		/* put the response code into the acc_param reason struct */
-		reason->code = t->uac[i].last_received;
-		reason->code_s.s = int2bstr((unsigned long)reason->code, code_buf, &reason->code_s.len);
+		if(reason!=NULL)
+		{
+			/* put the response code into the acc_param reason struct */
+			reason->code = t->uac[i].last_received;
+			reason->code_s.s = int2bstr((unsigned long)reason->code, code_buf, &reason->code_s.len);
+		}
 		/* get the contact from it */
 		n = shmcontact2dset( msg, t->uac[i].reply, max, reason, bflags);
 		if ( n<0 ) {
-- 
1.7.10.4

